examples/todo/src/Component/Task.purs

```diff
@@ -0,0 +1,82 @@
+module Example.Todo.Component.Task where
+
+import Prelude
+
+import Data.Maybe (Maybe(..))
+import Example.Todo.Model (Task, initialTask)
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+
+-- | The task component query algebra.
+data TaskQuery a
+  = IsCompleted (Boolean -> a)
+  | SetCompleted Boolean a
+
+-- | The messages the task component might send to a parent component.
+data TaskMessage
+  = NotifyRemove
+  | Toggled Boolean
+
+-- | The task component actions (arising from the rendered HTML).
+data TaskAction
+  = UpdateDescription String
+  | SetCompleted' Boolean
+  | Remove
```

Comment:
I used this todo example as my gateway to learning Halogen v5. The one thing I'm feeling uncertain on is how to deal with the situation where you want a query AND action for the same thing. In this case, that is "SetCompleted". I just duplicated the command and appended a ' in this case. This may or may not be idiomatic, I have no idea. It's not perfectly DRY, but neither is it terrible repetition either. Granted, it might not be terrible only because it's a tiny example.

Reply:
What I usually do is add a constructor to the query that can execute any action: `QueryAction TaskAction a`. Then you can call `handleAction` from `handleQuery` and run any action from the outside. But it looks like you do slightly different things in each case (raise a notification or not), so maybe you still want to keep them separate.

Reply:
This makes great sense. Will add to next commit.